Automattic/jetpack
# My Jetpack: Move VaultPress Backup back to hybrid product

Part of #29906. We are moving the products that have standalone plugins back to Hybrid products and then enhancing the status handling on them. This way we can mark them as active or not and push the standalone installation or not, on any given state.

## Proposed changes:

* Turn VaultPress Backup into a Hybrid product again, so we can manage it's status as a module and also as a standalone product
* Reintroduce unit testing on the `Hybrid_Product` class using VaultPress Backup as the testing subject

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Launch a new Jurassic Ninja site with only Jetpack
* Complete the Jetpack connection setup and choose to proceed with a free Jetpack plan
* Go to `Jetpack > My Jetpack`
* VaultPress will be shown as inactive and with a call to action to purchase (since it will be available through Jetpack but requires a plan to work), but will show as not absent (greyed):

<img width="350" alt="Screen Shot 2023-04-05 at 08 02 46" src="https://user-images.githubusercontent.com/6760046/230062209-7f17d1e4-a6b7-40b3-9229-ec40bbba0789.png">

* Click `Purchase` and follow the instructions until you finish the checkout
   * before this change, this flow would trigger the standalone plugin installation
   * now, this will enable the module without installing the standalone plugin; we plan to push the user to install the plugin after it, using a different button label and also an action on the Actions menu
* Confirm that the VaultPress Backup product shows as active and the `Manage` button is shown:

<img width="350" alt="Screen Shot 2023-04-04 at 19 21 17" src="https://user-images.githubusercontent.com/6760046/229935638-a06dd55f-d4b8-4561-be5e-d7a665459c62.png">

* Confirm that you are redirected to `cloud.jetpack.com/backup/yoursite.com?site=yoursite.com` when clicking the `Manage` button
* Confirm that the standalone plugin was not installed
* Go to `Jetpack > Jetpack Beta` and activate Jetpack Backup from `fix/backup-back-to-hybrid-product` feature branch
* Go back to `Jetpack > My Jetpack` and confirm that the VaultPress Backup product still shows as active
* Now disable and remove the Jetpack plugin to keep only the Jetpack Backup plugin
* Go to `Jetpack > My Jetpack` again and confirm that the VaultPress Backup product still shows as active
* Confirm that you are redirect to `admin.php?page=jetpack-backup` when clicking the `Manage` button
